### ISSUE TITLE
Bots have standard notification properties set and get notified via Email

### ISSUE THREAD

**Issue by apollo13** [mattermost/mattermost-server#15769]
#### Summary
Bot accounts get email notification delivered to `botaccountname@localhost`

#### Steps to reproduce
I am not 100% sure if this is actually the way to reproduce, but it seems simple enough:
 * Install the gitlab mattermost plugin and activate it
 * Open a private channel with the gitlab bot
 * Send a normal text message to it (no slash command)
 * Wait till the mail notification watcher triggers

#### Expected behavior
Bots should not get notifications

#### Observed behavior (that appears unintentional) 
I get a bounce notification from my mailserver since the `gitlab` bot received a notification for `gitlab@localhost`:
```
Auto-Submitted: auto-generated
Date: Mon, 05 Oct 2020 09:25:36 +0200
Precedence: bulk
From: "Mattermost" <mattermost@xxx>
To: gitlab@localhost
Subject: [GitLab Mattermost] New Notification for October 5, 2020
Content-Type: multipart/alternative;
 boundary=20a6a74cebc505baa3046ac6060fdf87d3f7a199cfea0debaa10b786c5be
```

#### Possible fixes
Disable notifications for bot accounts (Possibly via the `EnsureBot` api function)

**Comment by apollo13** [comment]
Please note that the notification is not triggered immediately (I saw a 15 minute delay or so).

**Comment by amyblais** [comment]
@apollo13 What Mattermost server version are you on?

**Comment by apollo13** [comment]
@amyblais Oh sorry, not sure how I missed that. I have whatever the gitlab omnibus installer installs, which currently is:
```
Mattermost Version: 5.26.2
Database Schema Version: 5.26.0
Database: postgres
```

**Comment by amyblais** [comment]
@DHaussermann Wondering if you can help test this?

**Comment by DHaussermann** [comment]
This seems quite possible but, I'm having a hard time confirming where I can see what's happening on the mail server.

Ideally what should happen when you create a bot (or convert a user to a bot), is that the email notification setting at the user level should be set to `never`. I'm not sure if this was ever implemented.

I'm trying to confirm this behavior on my local but I'm not sure where in the DB the user level email notification can be found.
@larkox maybe you have some suggestions on this?

**Comment by apollo13** [comment]
@DHaussermann Try the `users` table and the `notifyprops` field there. I fixed this yesterday by updating the table like this:
```
update users set notifyprops='{"channel":"true","comments":"never","desktop":"mention","desktop_sound":"false","email":"false","first_name":"false","mention_keys":"","push":"mention","push_status":"away"}'::json where email like '%@localhost';
```
The `email` key was true before my update.

**Comment by DHaussermann** [comment]
Thanks @apollo13  I saw that but it wasn't updating for me based on setting it in the UI (on a non-bot user). I'll take another look 👍

**Comment by apollo13** [comment]
@DHaussermann I just tested. The `email` key switches between `true/false` if you select "immediatelly"/"never" in the account settings -> notifications tab. But it will stay `true` also for "every hour" & "every 15 minutes" -- so that might be saved somewhere else.

**Comment by DHaussermann** [comment]
@apollo13 thanks for the help. I see the value change as expected. Somehow the query results must not have been refreshing for me when I checked earlier. 

You're correct about the email notification issue. It seems there is no exception made for bot accounts to prevent this. I have opened a Jira ticket here https://mattermost.atlassian.net/browse/MM-29374 to address this. Thank you for pointing this out. Much appreciated. 
cc @amyblais
